python/pipeline/behavior.py

```diff
@@ -63,4 +67,66 @@ def _make_tuples(self, key):
     def notify(self, key):
         msg = 'behavior.Sync for `{}` has been populated.'.format(key)
         (notify.SlackUser() & (experiment.Session() & key)).notify(msg)
+        
+@schema
+class Treadmill(dj.Computed):
+    definition = """
+    -> experiment.Scan
+    ---
+    treadmill_raw                       :longblob           #raw treadmill counts
+    treadmill_vel                       :longblob           #ball velocity integrated over 100ms bins in cm/sec 
+    treadmill_time                      :longblob           #timestamps of each sample in seconds on behavior clock
+    treadmill_ts = CURRENT_TIMESTAMP    :timestamp          #automatic
+    """
+    # adapted from Treadmill.m by Paul Fahey, 2017-10-13
+    def _make_tuples(self, key):
+        #pull filename for key
+        rel = experiment.Session() * experiment.Scan.BehaviorFile().proj(
+            hdf_file='filename')
+        info = (rel & key).fetch1()
+
+        # replace number by %d for hdf-file reader
+        tmp = info['hdf_file'].split('.')
+        if not '%d' in tmp[0]:
+            info['hdf_file'] = tmp[0][:-1] + '%d.' + tmp[-1]
+
+        #read hdf file for ball data
+        hdf_path = lab.Paths().get_local_path("{behavior_path}/{hdf_file}".format(**info))
+        data = read_video_hdf5(hdf_path)
+
+        #read out counter time stamp and convert to seconds
+        packet_length = data['analogPacketLen']
+        ball_time,_ = ts2sec(data['ball'].transpose()[1],packet_length)
+
+        #read out raw ball counts and integrate by 100ms intervals
+        ball_raw = data['ball'].transpose()[0]
+        ball_time_to_raw = interp1d(ball_time,ball_raw-ball_raw[0])
+        bin_times = np.arange(ball_time[0],ball_time[-1],.1)
+        bin_times[-1] = ball_time[-1]
+        ball_counts = np.append([0],np.diff(ball_time_to_raw(bin_times)))
+
+        #pull Treadmill specs, warn if more than one Treadmill fits session key
+        diam, counts_per_revolution = (
+            experiment.TreadmillSpecs() * experiment.Session() & key & 'treadmill_start_date <= session_date').fetch('diameter', 'counts_per_revolution')
+        if len(diam) != 1:
+            raise PipelineException('Unclear which treadmill fits session key')
+
+        #convert ball counts to cm/s for each ball time point
+        cmPerCount = np.pi*diam[-1]/counts_per_revolution[-1]
+        ball_time_to_vel = interp1d(bin_times, ball_counts*cmPerCount*10)
+        ball_vel = ball_time_to_vel(ball_time)
+
+        #assign calculated properties to key
+        key['treadmill_time'] = ball_time
+        key['treadmill_raw'] = ball_raw
+        key['treadmill_vel'] = ball_vel
+
+        #insert and notify user
+        self.insert1(key)
+        self.notify(key)
```

Comment:
small issue here is that the notification will have treadmill_time, treadmill_raw and treadmill_vel. One could do:
```python
tuple_ = key.copy()
tuple_['treadmill_time'] = treadmill_time
...
self.insert1(tuple_)
self.notify(key)
```